hibernate-reactive-core/src/main/java/org/hibernate/reactive/event/impl/DefaultReactiveLoadEventListener.java

```diff
@@ -361,7 +361,7 @@ else if ( persister.getRepresentationStrategy().getProxyFactory() != null ) {
 			// we have a HibernateProxy factory, this case is more complicated
 			return loadWithProxyFactory( event, persister, keyToLoad );
 		}
-		else if ( !persister.hasSubclasses() ) {
+		else if ( persister.hasSubclasses() ) {
```

Comment:
oops!

Reply:
Yeah :)

It was correct for ORM 6.4, but good luck noticing that it needs to be changed for ORM 6.5 without the proper test